### PR TITLE
Feat: fix/share rate simulation

### PR DESCRIPTION
**Pull request by TheDZhon**
- [x] Interface changes
- [x] Implementation
- [x] Don't break existing tests
- [ ] Test coverage to show why we need it
- [ ] Double-check burner test coverage

### CODE REVIEW

File: contracts/0.8.9/oracle/AccountingOracle.sol [hunk 1]
```diff
@@ -232,6 +233,10 @@ contract AccountingOracle is BaseOracle {
         /// at the reference slot.
         uint256 elRewardsVaultBalance;
 
+        /// @dev The shares amount requested to burn through Burner as observed
+        /// at the reference slot.
```
Comment (skozin) [inline contracts/0.8.9/oracle/AccountingOracle.sol:237]: maybe tell here how to obtain it? it's very impl-specific value anyway so referring the specific fn won't make the docs more fragile imo

File: contracts/0.8.9/Burner.sol [hunk 2]
```diff
@@ -312,9 +311,9 @@ contract Burner is IBurner, AccessControlEnumerable {
             coverSharesBurnRequested -= sharesToBurnNowForCover;
             sharesToBurnNow += sharesToBurnNowForCover;
         }
-        if ((memNonCoverSharesBurnRequested > 0) && (sharesToBurnNow < _sharesToBurnLimit)) {
+        if ((memNonCoverSharesBurnRequested > 0) && (sharesToBurnNow < _sharesToBurn)) {
```
Comment (skozin) [inline contracts/0.8.9/Burner.sol:314]: don't need inner parentheses here?

**Review by skozin — APPROVED** [review]

File: contracts/0.8.9/Burner.sol (hunk 2 above)
Comment (TheDZhon) [reply contracts/0.8.9/Burner.sol:314]: agree, removed :)
ty!

File: contracts/0.8.9/oracle/AccountingOracle.sol (hunk 1 above)
Comment (TheDZhon) [reply contracts/0.8.9/oracle/AccountingOracle.sol]: Yeah, added some more explanations :+1:

**Pull request merged**